sdks/java/core/src/main/java/org/apache/beam/sdk/io/DefaultFilenamePolicy.java

```diff
@@ -87,6 +130,17 @@ public static DefaultFilenamePolicy constructUsingStandardParameters(
   private final String shardTemplate;
   private final String suffix;
 
+  /*
+   * Checks whether given template contains enough information to form windowed file names
+   */
+  static boolean isWindowedTemplate(String template){
+    if (template != null){
+      Matcher m = WINDOWED_FORMAT_RE.matcher(template);
```

Comment:
I'm not sure this does what you want - it will be true if the template contains at least one of the parameters, right?